### PR TITLE
Fix a problem where masquerading as esm

### PR DESCRIPTION
**Pull request by himanoa**
Related: #191

Fix a problem Masquerading as ESM.

Changed to use tsup to generate `index.d.cts`. Accordingly, Created a tsup config.

### Operability confirmed

1. Exec `yarn workspace @rimbu/core run build`
2. Change directory to packages/core
3. Exec `npm pack`
4. Upload a generated tgz file to https://arethetypeswrong.github.io
5. Fixed masquerading as ESM problem

image

### All Submissions:

- [x] Have you followed the guidelines in our Contributing document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
6. [x] Have you lint and formatted your code locally prior to submission?

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by vitoke** [comment]
Impressive, nice work! So `tsup` is now used both for cjs and esm build right? Do you think this approach can be copied to all the other packages?

File: packages/core/package.json [hunk 1]
```diff
@@ -76,15 +84,12 @@
...
     "build:deno": "yarn bundle:deno-prepare && yarn bundle:deno-convert && yarn bundle:deno-move && yarn bundle:deno-clean",
-    "bundle": "yarn bundle:cjs && yarn bundle:esm && yarn bundle:types && yarn bundle:bun",
+    "bundle": "yarn tsup && yarn bundle:bun && yarn",
     "bundle:bun": "node ../../config/bunnify.mjs",
-    "bundle:cjs": "tsup src --format cjs --clean -d dist/cjs --loader '.mts=ts'",
     "bundle:deno-prepare": "node ../../config/prepare-denoify.mjs",
     "bundle:deno-convert": "denoify --src _deno_prepare/src",
     "bundle:deno-move": "rimraf ../../deno_dist/core && mv deno_dist ../../deno_dist/core",
     "bundle:deno-clean": "rimraf _deno_prepare",
-    "bundle:esm": "tsc --p tsconfig.esm.json",
```
Comment (vitoke) [inline packages/core/package.json:86]: Just to keep things clean, I think these json files can be removed too

**Comment by himanoa** [comment]
@vitoke 

> Impressive, nice work!
Thank you. ⭐

>  So tsup is now used both for cjs and esm build right? Do you think this approach can be copied to all the other packages?

Those two are correct. 👍 
Only the core is a little different from the other packages. so I think to change it in another Pull-Request. Is that OK?

File: packages/core/package.json (hunk 1 above)
Comment (himanoa) [reply packages/core/package.json:86]: I'll fixed it.

**Review by vitoke — APPROVED** [review]
Great improvement!

**Pull request merged**